### ISSUE TITLE
Change configuration after SDK init

### ISSUE THREAD

**Issue by ThomasKnobloch** [at-internet/piano-analytics-android#7]
Hello 👋 

We are facing an issue with the last version of the SDK 3.3.0.
The limitation that prevent us from changing the SDK configuration on-the-go is a major issue for us since we have multiple application environments and piano analytics dashboards (host & site) as well.

Can you provide a alternative to change configurations like on the previous versions ? (currently working on iOS)

Thank in advance

**Comment by DeKaN** [comment]
Hi, @ThomasKnobloch 

What do you want to achieve?
1. send some events for one host and site, change these settings, and send some other events
2. get host/site from some local/remote source and use them for events until app restart

**Comment by DeKaN** [comment]
Can you share a link to your app? Just want to see real case :)

**Comment by MehdiBrahmi** [comment]
@DeKaN our use case is :
We have a B2C application where a user can change his contextual region (French / Italy / Spain / ...)
Each region manages its own piano site (on the website https://explorer.atinternet-solutions.com/core)
When we used the version 2.x we could handle this configuration update but is not possible anymore with version 3.x

> **NOTE**: on iOS is possible with latest version.

**Comment by DeKaN** [comment]
Thanks for provided link, @MehdiBrahmi .

As I see the app reads flag `PIANO_ANALYTICS_ENABLED` from remote config and initializes only once PianoAnalytics with static values (domain and site id) in configuration and FirebaseAnalytics. Also, app pushes event only with two properties if the flag is true.

This app can be migrated to the latest PianoAnalytics without any problems.

**Comment by MehdiBrahmi** [comment]
Hi @DeKaN I think there's a misunderstanding about our request

The need is as follows :
As a user of Leroy Merlin application, I can change the country I want to shop in _(French, Italy, Spain, etc...)_

And for our business, we have a configuration by country like this sample:

'FR': { site: 112233, collectDomain: "[sample1234.domain.com](http://sample1234.domain.com/)" },
'IT': { site: 445566, collectDomain: "[sample5678.domain.com](http://sample5678.domain.com/)" },
'ES': { site: 778899, collectDomain: "[sample9012.domain.com](http://sample9012.domain.com/)" },

> **NOTE:** it's a business choice, I can't change the way it works.

So when a user changes country in our application _(on this screen)_
<img src="https://github.com/at-internet/piano-analytics-android/assets/25221326/4ed9fe48-f381-40b0-997b-d046ec97f7fd" width=300px />

We need to re-initialise `PianoAnalytics` with the configuration according to the new selected country _(as described above)_

Currently, when you initialise the SDK by following the documentation, you have to do the following:

```kotlin
val configuration = Configuration.Builder(
    collectDomain = selectedCountry.piano.collectDomain,
    site = selectedCountry.piano.site
).build()

PianoAnalytics.init(applicationContext, configuration)
```

So we're replaying this code for each change of country but if i check inside the code of Piano SDK, i can see this:

```kotlin

        @JvmStatic
        @Volatile
        private var instance: DependenciesProvider? = null

        @JvmStatic
        internal fun init(context: Context, configuration: Configuration, dataEncoder: DataEncoder) {
            if (instance == null) {
                synchronized(this) {
                    if (instance == null) {
                        instance = DependenciesProvider(context.applicationContext, configuration, dataEncoder).also {
                            Thread.setDefaultUncaughtExceptionHandler(it.crashHandler)
                        }
                    }
                }
            }
        }
```

This seems to mean that it is not possible (in this version 3.3.x) to update the configuration according to the country selected.

So the request is :
Is it possible to have _(like before on the [v3.2.1](https://github.com/at-internet/piano-analytics-android/tree/v3.2.1))_ a method like :

```kotlin
PianoAnalytics.getInstance().setConfiguration(
        Configuration.Builder(
                collectDomain = selectedCountry.piano.collectDomain,
                site = selectedCountry.piano.site
        ).build()
)
```

Thank you

**Comment by DeKaN** [comment]
Hi @MehdiBrahmi, I think for your case may be useful something like 
```
interface ReportUrlProvider {
    val collectDomain: String
    val site: Int
    val path: String
        get() = Configuration.DEFAULT_PATH
}
```
But you can get "incorrect" events for this case:
1. you called `sendEvents`, SDK saved them and got an error at sending
2. you change `collectDomain`/`site`
3. now all events (including not sent before) will be sent to new `collectDomain`/`site`

**Comment by MehdiBrahmi** [comment]
Hi @DeKaN, thanks for your reply. Is it possible to clean up saved events _(with a method we can call)_ when we need to change our collectDomain / site?

**Comment by DeKaN** [comment]
Hi @MehdiBrahmi,
You can call [`deleteOfflineStorage()`](https://github.com/at-internet/piano-analytics-android/blob/main/piano-analytics/src/main/java/io/piano/android/analytics/PianoAnalytics.kt#L151) to drop all saved events

**Comment by MehdiBrahmi** [comment]
Hi @DeKaN, thank you for adding this feature

**Comment by DeKaN** [comment]
Released with version 3.3.2
